source/s3_auto_ranged_put.c

```diff
@@ -1298,19 +1289,13 @@ static struct aws_future *s_s3_prepare_complete_multipart_upload(struct aws_s3_r
 static void s_s3_prepare_complete_multipart_upload_on_skipping_done(void *user_data) {
 
     struct aws_s3_prepare_complete_multipart_upload_async_ctx *complete_mpu_prep = user_data;
-    struct aws_s3_request *request = complete_mpu_prep->request;
-    struct aws_s3_meta_request *meta_request = request->meta_request;
-    struct aws_s3_auto_ranged_put *auto_ranged_put = meta_request->impl;
 
     int error_code = aws_future_get_error(complete_mpu_prep->skipping_future);
     if (error_code != AWS_ERROR_SUCCESS) {
         s_s3_prepare_complete_multipart_upload_finish(complete_mpu_prep, error_code);
         return;
     }
 
-    /* Skipping was successful */
-    auto_ranged_put->prepare_data.num_parts_read_from_stream = auto_ranged_put->synced_data.total_num_parts;
```

Comment:
Couple of options:
- Use size of etag list
- Only update it if content_length is available
- since num_parts_read_from_stream is only used for skipping parts, we can also just not update it.

Reply:
let's just give this a name indicating it's only used for skipping. hard to think of a good one:
part_index_for_skipping
skipping_progress
skipping_resumes_at

Reply:
Thanks, updated it to part_index_for_skipping.